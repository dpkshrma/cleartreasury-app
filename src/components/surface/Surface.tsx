import * as React from "react";

const Align = {
  LEFT: "justify-start",
  RIGHT: "justify-end",
  CENTER: "justify-center",
} as const;

type StaticProperties = {
  Align: typeof Align;
};

type Values<T> = T[keyof T];

type Props = {
  children: React.ReactNode;
  layoutClass?: string;
  align?: Values<typeof Align>;
};

const Surface: React.FC<Props> & StaticProperties = ({
  align = Align.LEFT,
  children,
}) => <div className={`flex ${align} w-full py-10 bg-white`}>{children}</div>;

Surface.Align = Align;

export default Surface;
