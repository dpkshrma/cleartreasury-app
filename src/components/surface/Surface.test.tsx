import * as React from "react";
import { render } from "../../../test/testUtils";
import * as Stories from "./Surface.stories";

describe("Surface", () => {
  it("matches snapshot for Default", () => {
    const { asFragment } = render(<Stories.Default />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for Center", () => {
    const { asFragment } = render(<Stories.Center />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for Right", () => {
    const { asFragment } = render(<Stories.Right />);
    expect(asFragment()).toMatchSnapshot();
  });
});
