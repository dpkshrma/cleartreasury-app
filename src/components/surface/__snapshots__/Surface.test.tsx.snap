// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surface matches snapshot for Center 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center w-full py-10 px-4 bg-theme-color-surface"
  >
    <div
      class="max-w-xl w-full"
    >
      Children here
    </div>
  </div>
</DocumentFragment>
`;

exports[`Surface matches snapshot for Default 1`] = `
<DocumentFragment>
  <div
    class="flex justify-start w-full py-10 px-4 bg-theme-color-surface"
  >
    <div
      class="max-w-xl w-full"
    >
      Children here
    </div>
  </div>
</DocumentFragment>
`;

exports[`Surface matches snapshot for Right 1`] = `
<DocumentFragment>
  <div
    class="flex justify-end w-full py-10 px-4 bg-theme-color-surface"
  >
    <div
      class="max-w-xl w-full"
    >
      Children here
    </div>
  </div>
</DocumentFragment>
`;
